src/Cache.ts

```diff
@@ -231,6 +286,9 @@ export class Cache {
         const tasks: Task[] = [];
         const fileLines = fileContent.split('\n');
 
+        // lazily store date extracted from filename. The value is undefined until the filename has been parsed.
+        let dateFromFileName: Moment | null | undefined = undefined;
```

Comment:
The difference was that I needed the DateWrapper so I could update the date from a closure because I couldn't modify the date directly.

I refactored this using a `Lazy` class which better captures the intention and hides the technicalities of the lazy implementation from this level of abstraction.